### PR TITLE
Refactor pkg/restore tests (part 2)

### PR DESCRIPTION
**Pull request by skriss**
Again, probably easiest to review one commit at a time.

I found some bugs in the code that determines which restore item actions to run - I omitted the test cases that failed from this PR, but will be submitting another PR with the bug fix and the test cases that cover them.

@nrb @prydonius @carlisia

### CODE REVIEW

**Comment by skriss** [comment]
found some more obsolete code, amended the last commit to remove it

File: pkg/restore/restore_new_test.go [hunk 1]
```diff
@@ -955,6 +958,173 @@ func TestRestoreItems(t *testing.T) {
...
+			name:    "actions with selectors that don't match anything don't run for any resources",
+			restore: defaultRestore().Restore(),
+			backup:  defaultBackup().Backup(),
+			tarball: newTarWriter(t).
+				addItems("pods", test.NewPod("ns-1", "pod-1")).
+				addItems("persistentvolumeclaims", test.NewPVC("ns-2", "pvc-2")).
+				done(),
+			apiResources: []*test.APIResource{test.Pods(), test.PVCs(), test.PVs()},
+			actions: map[*recordResourcesAction][]string{
+				new(recordResourcesAction).ForNamespace("ns-1").ForResource("persistentvolumeclaims"): nil,
```
Comment (nrb) [inline pkg/restore/restore_new_test.go]: Should these values be `nil` or `""`?

File: pkg/restore/restore_new_test.go [hunk 2]
```diff
@@ -721,6 +724,237 @@ func TestInvalidTarballContents(t *testing.T) {
...
+				test.Pods(),
+			},
+			want: []*test.APIResource{
+				test.Pods(
+					test.NewPod("ns-1", "pod-1", test.WithLabels("velero.io/backup-name", "backup-1", "velero.io/restore-name", "restore-1")),
+				),
+			},
+		},
+		{
+			name:    "object gets labeled with full backup and restore names when they're both longer than 63 characters",
```
Comment (nrb) [inline pkg/restore/restore_new_test.go]: Isn't this the shorter case, not the longer one?

Comment (skriss) [reply pkg/restore/restore_new_test.go]: yep, will update

File: pkg/restore/restore_new_test.go (hunk 1 above)
Comment (skriss) [reply pkg/restore/restore_new_test.go]: they should be nil, since the action's `ids` field should never be initialized/should always stay nil

Comment (nrb) [reply pkg/restore/restore_new_test.go]: Ah, I see my confusion - it's a `nil` slice of strings, not a string.

**Comment by skriss** [comment]
@prydonius would be great if you could review this today - I have a few other PRs backing up waiting for this to merge. Thanks!

**Pull request merged**